django/db/models/constraints.py

```diff
@@ -89,6 +90,12 @@ def clone(self):
         _, args, kwargs = self.deconstruct()
         return self.__class__(*args, **kwargs)
 
+    def only_ddl_impacted_fields(self):
+        mutable_self = copy.copy(self)
+        mutable_self.violation_error_code = None
+        mutable_self.violation_error_message = None
+        return mutable_self
```

Comment:
Hi @adriennefranke, first timer on django core - hopped here from the djangocon US talks.

Thanks for the patch, and think its really an interesting one following the trend of conversations from #17644 and #13488.

In addition, to @charettes, could I suggest sending in a discourse update on the regarding this patch to discord as well?

Reply:
@50-Course Yes, I will do that as well. Thanks!